Shared/Samples/Generate offline map with local basemap/GenerateOfflineMapWithLocalBasemapView.swift

```diff
@@ -0,0 +1,294 @@
+                                // Creates a rectangle from the area of interest.
+                                let viewRect = geometry.frame(in: .local).inset(
+                                    by: UIEdgeInsets(
+                                        top: 20,
+                                        left: geometry.safeAreaInsets.leading + 20,
+                                        bottom: 44,
+                                        right: -geometry.safeAreaInsets.trailing + 20
+                                    )
+                                )
+                                
+                                // Creates an envelope from the rectangle.
+                                guard let extent = mapView.envelope(fromViewRect: viewRect) else { return }
+                                
+                                do {
+                                    // Generates an offline map.
+                                    try await model.generateOfflineMap(extent: extent)
+                                } catch {
+                                    self.error = error
+                                }
+                                
+                                // Sets generating an offline map to false.
+                                isGeneratingOfflineMap = false
+                            }
+                        }
+                    }
+            }
+        }
+    }
+}
+
+private extension GenerateOfflineMapWithLocalBasemapView {
+    /// The model used to store the geo model and other expensive objects
+    /// used in this view.
+    @MainActor
+    class Model: ObservableObject {
+        /// The offline map that is generated.
+        @Published private(set) var offlineMap: Map!
+        
+        /// A Boolean value indicating whether the generate button is disabled.
+        @Published private(set) var isGenerateDisabled = true
+        
+        /// A Boolean value indicating if the local basemap is available to be used.
+        @Published var canUseLocalBasemap: Bool = {
+            FileManager.default.fileExists(atPath: URL.napervilleImageryBasemap.path())
+        }()
+        
+        /// A Boolean value indicating whether the offline map uses the local or online basemap.
+        @Published var usesLocalBasemap = false
+        
+        /// The generate offline map job.
+        @Published private(set) var generateOfflineMapJob: GenerateOfflineMapJob!
+        
+        /// The offline map task.
+        private var offlineMapTask: OfflineMapTask!
+        
+        /// A URL to a temporary directory where the offline map files are stored.
+        private let temporaryDirectory = createTemporaryDirectory()
+        
+        /// A portal item displaying the Naperville, IL water network.
+        private let napervillePortalItem = PortalItem(
+            portal: .arcGISOnline(connection: .anonymous),
+            id: PortalItem.ID("acc027394bc84c2fb04d1ed317aac674")!
+        )
```

Comment:
Not a big deal, but I would consider moving this into closure that creates the `offlineMap` if it is only used for that. You can also move the `onlineMap.minScale = 1e4` there and also get rid of the model's initializer. Up to you, though.